common/src/main/java/com/wynntils/models/gear/GearModel.java

```diff
@@ -80,24 +86,44 @@ public boolean canBeGearBox(GearInfo gear) {
 
     @Override
     public void reloadData() {
-        gearInfoRegistry.reloadData();
+        gearInfoRegistry.loadData();
     }
 
+    // For "real" gear items eg. from the inventory
     public GearInstance parseInstance(GearInfo gearInfo, ItemStack itemStack) {
         WynnItemParseResult result = WynnItemParser.parseItemStack(itemStack, gearInfo.getVariableStatsMap());
         if (result.tier() != gearInfo.tier()) {
             WynntilsMod.warn("Tier for " + gearInfo.name() + " is reported as " + result.tier());
         }
 
+        SetInstance setInstance = null;
+        if (result.tier() == GearTier.SET && gearInfo.setInfo().isPresent()) {
+            SetInfo setInfo = gearInfo.setInfo().get();
+            setInstance = new SetInstance(setInfo, getActiveItems(setInfo.name()), getWynncraftCount(setInfo.name()));
```

Comment:
Overall, it feels like that this is still misplaced. I think it should be `WynnItemParser` parsing the "wynncraft count", **along with parsing the equipped items from the tooltip**. I know, it won't be "accurate" for the duplication case, but it shouldn't be! Here, we are parsing what Wynn tells us..., and eventually, it will be correct.

So, I think, you should just parse and include the SetInstance in `WynnItemParseResult`. That would also mean that `getActiveItems` and `getWynncraftCount` can basically go, and all parsing is done with regexes.